src/main/java/com/gooddata/executeafm/afm/OverPeriodDateAttribute.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2007-2018, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata.executeafm.afm;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.gooddata.executeafm.ObjQualifier;
+import com.gooddata.util.GoodDataToStringBuilder;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+import static org.apache.commons.lang3.Validate.notNull;
+
+/**
+ * Definition of the {@link OverPeriodMeasureDefinition} attribute.
+ */
+public class OverPeriodDateAttribute implements Serializable {
+
+    private static final long serialVersionUID = 2311364644023464059L;
+
+    private final ObjQualifier attribute;
+    private final Integer periodsAgo;
+
+    /**
+     * Create a new instance of {@link OverPeriodDateAttribute}.
+     *
+     * @param attribute
+     *         The {@link ObjQualifier} of the attribute from the date data set that defines the PoP period and date data set. The parameter must not be
+     *         null.
+     * @param periodsAgo
+     *         The number of periods defined by the {@code attribute} about which this period will be shifted about. The positive number shifts the period to
+     *         the past, the negative to the future. The parameter must not be null.
+     *
+     * @throws NullPointerException
+     *         Thrown when one of the required parameter is null.
+     */
+    @JsonCreator
+    public OverPeriodDateAttribute(
+            @JsonProperty("attribute") final ObjQualifier attribute,
+            @JsonProperty("periodsAgo") final Integer periodsAgo) {
+        this.attribute = notNull(attribute, "'attribute' must not be null!");
```

Comment:
please use https://github.com/gooddata/gooddata-rest-common/blob/master/src/main/java/com/gooddata/util/Validate.java#L24

as it's used everywhere among gooddata-java

Reply:
Fixed. I did not noticed that.